### PR TITLE
CAPA: Release v29.0.0.

### PR DESCRIPTION
**Pull request by Gacko**
### Checklist

- [x] Roadmap issue created
- [x] Release uses latest stable Flatcar
- [x] Release uses latest Kubernetes patch version

### Triggering E2E tests

To trigger the E2E test for each new Release added in this PR, add a comment with the following:

`/run releases-test-suites`

If you want to trigger conformance tests, you can do so by adding a comment similar to the following:

`/run conformance-tests PROVIDER=capa RELEASE_VERSION=v29.0.0`

For more details see the [README.md](/README.md#running-tests-against-prs).

### CODE REVIEW

**Review by AndiDog — COMMENTED** [review]
Change looks fine, apart from test failures. LGTM.

File: capa/v29.0.0/release.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    version: 0.1.1
+    catalog: cluster
+    dependsOn:
+    - cilium
+  - name: node-exporter
+    version: 1.19.0
+    dependsOn:
+    - kyverno
+  - name: observability-bundle
+    version: 1.3.4
```
Comment (TheoBrigitte) [inline capa/v29.0.0/release.yaml:91]: Is it possible to update `observability-bundle` to 1.5.0 in this PR or should we wait for next v29.x.x ?

```suggestion
    version: 1.5.0
```

Comment (Gacko) [reply capa/v29.0.0/release.yaml:91]: Sure, it is, but for now I'd like to get the cluster running at all. I'll leave a message when it's ready for new app version. :)

**Comment by fhielpos** [comment]
I would also like to do a bundle update when possible 😄 

https://github.com/giantswarm/releases/pull/1319

We are splitting Kyverno and its CRDs, so the `dependsOn` should also be updated.

**Comment by Gacko** [comment]
@giantswarm/team-tinkerers It seems conformance tests are stuck / not coming up properly. The cluster looks good, but the conformance tests itself stay at `Completed = 0`. Any idea?

File: capa/v29.0.0/release.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  - name: cert-manager
+    version: 3.7.6
+    dependsOn:
+    - prometheus-operator-crd
+  - name: chart-operator-extensions
+    version: 1.1.2
+    dependsOn:
+    - prometheus-operator-crd
+  - name: cilium
+    version: 0.24.0
```
Comment (ubergesundheit) [inline capa/v29.0.0/release.yaml:34]:
```suggestion
    version: 0.25.1
```

Please upgrade cilium-app to 0.25.1 (CAPA 25.1.0 will also have this version)
The changes between 0.24.0 and 0.25.1 are just a cilium patch bump to 1.15.6 and said BPF map size regression

thanks

Comment (mcharriere) [reply capa/v29.0.0/release.yaml:34]: https://github.com/giantswarm/releases/pull/1323

I created that PR including all Cabbage apps

Comment (Gacko) [reply capa/v29.0.0/release.yaml:34]: Thank you, Mati! This is how we had it in mind!

File: capa/v29.0.0/release.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    version: 5.2.2
+    dependsOn:
+    - prometheus-operator-crd
+  - name: vertical-pod-autoscaler-crd
+    version: 3.1.0
+  components:
+  - name: cluster-aws
+    catalog: cluster
+    version: 1.2.1
+  - name: flatcar
```
[the remark this answers is not in the document]
Comment (Gacko) [reply capa/v29.0.0/release.yaml:115]: The variant renders the image lookup format with a `alpha` suffix. Right now, there is no such image and the latest image without that suffix is still up-to-date. tl;dr: No variant needed for now, but @nprokopic already filed [an issue](https://github.com/giantswarm/roadmap/issues/3592) for @giantswarm/team-tinkerers  to improve the situation around this. 🙂

**Review by stone-z — CHANGES_REQUESTED** [review]
Please to include the [security-bundle update](https://github.com/giantswarm/releases/pull/1319)

**Comment by Gacko** [comment]
Thank you, @stone-z! I didn't want to merge it, but requesting a change is a good idea to actively prevent this and I cannot do this as the creator of the PR. 😅

**Comment by Gacko** [comment]
The current [CNCF Conformance Tests run](https://tekton.ci.giantswarm.io/#/namespaces/tekton-pipelines/pipelineruns/pr-releases-1318-conformance-testskmrm4) is more or less passing as the failing tests are known to fail due to Cilium and Kyverno.

File: capa/v29.0.0/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+### cilium [v0.25.1](https://github.com/giantswarm/cilium-app/releases/tag/v0.25.1)
+
+#### Changed 
+
+- Fix regression setting Policy BPF Max map `policyMapMax` back to 65536 from 16384.
+- Upgrade cilium to v1.15.6.
+
+### k8s-dns-node-cache [v2.8.1](https://github.com/giantswarm/k8s-dns-node-cache-app/releases/tag/v2.8.1)
+
+### Changed
```
Comment (mcharriere) [inline capa/v29.0.0/README.md:20]:
```suggestion
#### Changed
```

File: capa/v29.0.0/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  - Enable readOnly FS moving config to emptyDir volume.
+  - Remove `NET_ADMIN` and drop `ALL` capabilities.
+  - Add `NET_BIND_SERVICE` capability.
+  - Add policy exception for `require-non-root-groups/autogen-check-runasgroup`.
+  - Remove disallow-capabilities-* policy exceptions.
+- Update PolicyException CR version to v2beta1.
+
+### net-exporter [v1.21.0](https://github.com/giantswarm/net-exporter/releases/tag/v1.21.0)
+
+### Changed
```
Comment (mcharriere) [inline capa/v29.0.0/README.md:32]:
```suggestion
#### Changed
```

File: capa/v29.0.0/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+- cluster-aws from 1.0.1 to 1.2.1
+- Flatcar from 3815.2.2 to 3815.2.5
+- Kubernetes from 1.28.11 to 1.29.7
+- cloud-provider-aws (formerly aws-cloud-controller-manager-app) from 1.28.6-gs1 to 1.29.3-gs1
+- cluster-autoscaler from 1.28.5-gs1 to 1.29.3-gs1
+
+### cilium [v0.25.1](https://github.com/giantswarm/cilium-app/releases/tag/v0.25.1)
+
+#### Changed
```
Comment (mcharriere) [inline capa/v29.0.0/README.md:13]:
```suggestion
#### Changed
```

**Review by stone-z — APPROVED** [review]

**Comment by Gacko** [comment]
inb4: Since release test suites failed to create a cluster for the upgrade tests yesterday, I just created a v28.1.0 manually and that worked pretty fine. As this is also the planned upgrade path, I'll now trigger another run with v28.1.0 explicitly specified as previous version.

**Comment by Gacko** [comment]
Ok, well, docs are wrong about the `PREVIOUS_RELEASE` argument. 😅

**Comment by AverageMarcus** [comment]
> Ok, well, docs are wrong about the `PREVIOUS_RELEASE` argument. 😅

They're not 🤨 That is the correct arg to pass in. What makes you say that?

**Comment by AverageMarcus** [comment]
@Gacko It looks like the previous version now needs to be provided as just the number version, not with the `aws-v` prefix 🤔  Not sure when that changed.

**Comment by Gacko** [comment]
Currently [the documentation of this repository here](https://github.com/giantswarm/releases/blob/master/README.md?plain=1#L474), so none of "yours", states as follows:

> The upgrade tests will first install a cluster using the latest previously published release and then upgrade to using the release found in the PR. If you need to override the starting Release version (to test upgrading from an earlier release) you can do so by setting the PREVIOUS_RELEASE parameter on the trigger comment (e.g. /run releases-test-suites PREVIOUS_RELEASE=aws-v25.0.0)

With that format, [the pipeline](https://tekton.ci.giantswarm.io/#/namespaces/tekton-pipelines/pipelineruns/pr-releases-1318-releases-test-suitesq5qt7) fails for the following reason:

```
[FAILED] Unexpected error:
    <*errors.errorString | 0xc0009ec160>: 
    failed to build cluster app: GET https://api.github.com/repos/giantswarm/releases/contents/capa/vaws-v28.1.0/release.yaml?ref=efb86ca3d8534d647299f9084d0fb6f6d984dfb6: 404 Not Found []
    {
        s: "failed to build cluster app: GET https://api.github.com/repos/giantswarm/releases/contents/capa/vaws-v28.1.0/release.yaml?ref=efb86ca3d8534d647299f9084d0fb6f6d984dfb6: 404 Not Found []",
    }
occurred
In [BeforeSuite] at: /go/pkg/mod/github.com/giantswarm/cluster-standup-teardown@v1.18.0/pkg/standup/standup.go:75
```

See the release name in the URL: `https://api.github.com/repos/giantswarm/releases/contents/capa/vaws-v28.1.0/release.yaml`.

**Comment by Gacko** [comment]
Ah damn, you were faster. :P

Overall the formats are a bit... inconsistent. But I really don't wanna complain! I barely use these arguments and when I do, I normally look them up in our examples. So no need to change them now.

**Comment by Gacko** [comment]
> @Gacko It looks like the previous version now needs to be provided as just the number version, not with the aws-v prefix 🤔 Not sure when that changed.

This absolutely makes sense to me as you probably don't wanna test a whole different provider, but only a different version of the same provider. So calling it `PREVIOUS_RELEASE_VERSION` would probably be a bit clearer.

Whereas it's totally fine to call the argument `TARGET_RELEASES` for defining only specific releases to test as there can be new releases for multiple providers in one PR and you want to be clear about the provider and version to test.

**Comment by AverageMarcus** [comment]
>  so none of "yours", states as follows:

I wrote those docs! 😝 

I think it'd be best if we just support both types. If there is a provider prefix we just strip it off and ignore it.

**Comment by AverageMarcus** [comment]
Oh actually, we do try and handle this but are hit AGAIN by the capa vs aws inconsistency 🤦 

https://github.com/giantswarm/clustertest/blob/288d57d784de612f11464a83264f8036930fa2c4/pkg/application/cluster_app.go#L180

**Comment by Gacko** [comment]
@AverageMarcus: Another question. The following line is from [a failed upgrade test suite](https://tekton.ci.giantswarm.io/#/namespaces/tekton-pipelines/pipelineruns/pr-releases-1318-releases-test-suites7g9hg):

```
{"level":"info","ts":"2024-08-07T13:16:46Z","msg":"Release name: 'aws-29.0.0-t.q4mwz4wjod'"}
```

I triggered this run without the `PREVIOUS_RELEASE` argument, so it should determine the version on its own. As it was an upgrade test, I'd expect the release name to be something like `aws-28.1.0`. Is this suite actually picking `aws-28.1.0` and just naming it like this or is the implementation faulty?

**Comment by Gacko** [comment]
Another try without defining the previous release version:

/run releases-test-suites

**Pull request merged**